tests/test_book_model.py

```diff
@@ -55,4 +56,56 @@ def test_to_dict_missing_description():
     assert len(result) == 3
     assert result["id"] == 1
     assert result["title"] == "Ocean Book"
-    assert result["description"] is None
\ No newline at end of file
+    assert result["description"] is None
+
+def test_from_dict_returns_book():
+    # Arrange
+    book_data = {
+        "title": "New Book",
+        "description": "The Best!"
+    }
+
+    # Act
+    new_book = Book.from_dict(book_data)
+
+    # Assert
+    assert new_book.title == "New Book"
+    assert new_book.description == "The Best!"
+
+def test_from_dict_with_no_title():
+    # Arrange
+    book_data = {
+        "description": "The Best!"
+    }
+
+    # Act & Assert
+    with pytest.raises(KeyError, match = 'title'):
+        new_book = Book.from_dict(book_data)
+
+def test_from_dict_with_no_description():
+    # Arrange
+    book_data = {
+        "title": "New Book"
+    }
+
+    # Act & Assert
+    with pytest.raises(KeyError, match = 'description'):
+        new_book = Book.from_dict(book_data)
+
+def test_from_dict_with_extra_keys():
+    # Arrange
+    book_data = {
+        "extra": "some stuff",
+        "title": "New Book",
+        "description": "The Best!",
+        "another": "last value"
+    }
+    
+    # Act
+    new_book = Book.from_dict(book_data)
+
+    # Assert
+    assert new_book.title == "New Book"
+    assert new_book.description == "The Best!"
+    assert hasattr(new_book, "extra") == False
+    assert hasattr(new_book, "another") == False
```

Comment:
Fancy. And while I do like how thorough this is (see? no extra property got added to the book) I wonder if this invites more potential confusion. Like I wouldn't write this code in a real test, since of course just adding a key to the dictionary doesn't dynammically cause a new property to appear in our class that didn't define that property. (It's a little wishy-washy to say that in python, where technically it _could_ happen, but once we open that can of worms, where does it stop?)

Reply:
Specifically because we're in Python where if someone made a typo, it can add a new attribute to an object, I wanted to add this test and assert. In my mind, it's simple enough to frame as "In python objects can have new attributes added to them that aren't defined in the class, so this test is to ensure we didn't have any typos in our `from_dict` function . In many other languages this is not possible and we wouldn't need a test for this scenario." but if there's an aspect I'm missing to your concern or the consensus is that it's potentially confusing overall I'm good removing it.

Reply:
Our `from_dict` is still relying on the `Model`-supplied ctor, which won't add non-column values to the instance (it raises TypeError if we give it an unexpected column). I think for me, it feels like a non-sequitor to bring up the possiblity of unintended additional fields here for the first time.

Reply:
That's fair, I'll cut the `hasattr` asserts in these tests for each of the models as they come up in the branches